server/src/main/java/org/cloudfoundry/identity/uaa/oauth/beans/OauthEndpointBeanConfiguration.java

```diff
@@ -550,11 +551,17 @@ CurrentUserCookieRequestFilter currentUserCookieFilter(
         return new CurrentUserCookieRequestFilter(currentUserCookieFactory);
     }
 
+    @Bean
+    RestTemplateConfig restTemplateConfig() {
```

Comment:
Does this not conflict with 
```
@Configuration
public class RestTemplateConfig
```

Should this be a `@Configuration` or a `@Bean` but maybe not both?

Reply:
What happens if you do

```
    @Autowired
    RestTemplateConfig restTemplateConfig
```

Reply:
> @Autowired
>     RestTemplateConfig restTemplateConfig

Tested, but without extra bean configuration for the `@Configuration` I get a singleton... so it is the same ref.

To me we can also do this because at least we would have again a RestTemplete with Pooled connections... as we had before..

If you compare to 77.20.x and before then ok.

If you compare a 76.31.0 UAA with ours then as mentioned, because of extra bean configuration in saml-idp.xml the call to the method create always new Resttemplates and Pools (maybe also not that what we want)

Reply:
Please check https://github.com/cloudfoundry/uaa/pull/3434 . 

I think it is better to go with https://github.com/cloudfoundry/uaa/pull/3434 and stay with using the beans for RestTemplates.

In debugger I see that SamlConfig beans e.g. nonTrustingRestTemplate wins in start and internally with HttpClient there is a default pooled connection manager created, but only with the default of 10 (all) and 5 (per route). And that would explain our issue in large systems with huge load.